File: Hashing/max_frequency.cpp
#include <iostream>
#include <vector>
#include <map>
using namespace std;

class Solution
{
public:
    int mostFrequentElement(vector<int> &nums)
    {
        map<int,int> hash;
        int max_value = 0;
        int max_element;

        // Hashing
        for (auto &e : nums)
        {
            hash[e]++;
        }

        // Comparing
        for (auto& i : hash)
        {
            if (i.second > max_value)
            {
                max_value = i.second;
                max_element = i.first;
            }
        }

        return max_element;
    }
};

int main() {
    vector<int> nums = {10,9,7};
    Solution sol;
    int result = sol.mostFrequentElement(nums);
    cout << "Most frequent element: " << result << endl;
    return 0;
}

File: Hashing/max_frequency.py
class Solution:
    def mostFrequentElement(self, nums):
        hash = {}

        for i in nums:
            hash[i] = hash.get(i,0) + 1

        max_value = max(hash.values())

        max_element = [k for k,v in hash.items() if(v == max_value)]

        return min(max_element)

if __name__ == "__main__":
    nums = [10,9,7]
    sol = Solution()
    print(sol.mostFrequentElement(nums))